main.tf

```diff
@@ -11,32 +25,27 @@ resource "aws_acm_certificate" "default" {
 }
 
 locals {
-  fqdn                    = var.hostname != "" ? format("%s.%s", var.hostname, var.domain) : var.domain
-  san                     = formatlist("%s.%s", var.subject_alternative_names, var.domain)
-  skip_route53_validation = var.validation_method == "EMAIL" ? true : var.skip_route53_validation
-  create_route53_record   = var.create_route53_record && false == local.skip_route53_validation
-}
-
-data "aws_route53_zone" "default" {
-  count = local.create_route53_record ? 1 : 0
-
-  name         = "${var.domain}."
-  private_zone = false
+  dvo_data = (! local.create_route53_record) ? {} : {
+    for dvo in aws_acm_certificate.default.domain_validation_options : dvo.domain_name => {
+      name   = dvo.resource_record_name
+      record = dvo.resource_record_value
+      type   = dvo.resource_record_type
+    }
+  }
 }
 
 resource "aws_route53_record" "default" {
-  count = local.create_route53_record ? length(var.subject_alternative_names) + 1 : 0
-
-  name    = aws_acm_certificate.default.domain_validation_options[count.index]["resource_record_name"]
-  type    = aws_acm_certificate.default.domain_validation_options[count.index]["resource_record_type"]
-  zone_id = data.aws_route53_zone.default[0].id
-  ttl     = 60
-  records = [aws_acm_certificate.default.domain_validation_options[count.index]["resource_record_value"]]
+  for_each = local.dvo_data
```

Comment:
@JonRoma does `dvo.domain_name` line up with `var.subject_alternative_names` if so then why not try `for_each = var.subject_alternative_names`? Just a random thought. Not sure if it will work.